dockerfile/cosmos/local.Dockerfile

```diff
@@ -8,12 +8,18 @@ ARG BUILDARCH
 ARG GITHUB_ORGANIZATION
 ARG REPO_HOST
 ARG GITHUB_REPO
+ARG BUILD_DIR
+ARG BUILD_TARGET
 
 WORKDIR /go/src/${REPO_HOST}/${GITHUB_ORGANIZATION}/${GITHUB_REPO}
 
+ADD . .
```

Comment:
Moving the `ADD` up here removes all of the caching optimizations that were done for local builds in #107 . Is there a reason it's necessary?

Reply:
The optimization did not account for a build dir. When a build dir is defined, we would need to pull the go.mod and go.sum from that build directory for the wasmvm search, not from the root. ADD would not pull from the build directory as expected and I don't think we want to put those files in root when that is not their expected location. We can re-add the optimization if we find a way to copy the build dir go.mod/sum files to their build dir location, but the optimization currently breaks building when there is a build dir.

Reply:
What if we just move up the `ARG BUILD_DIR` so that it is available when adding the go.mod and go.sum, and then `ADD ${BUILD_DIR}/go.mod ${BUILD_DIR}/go.sum ./` ? 

Alternatively, what if we move the wasm version detection into the heighliner go code and add `ARG WASM_VERSION` to the dockerfiles? Heighliner already parses the go.mod to determine the go version. That would probably help caching the most so that libwasm wouldn't need to be re-downloaded when changing the go.mod, only when the wasm version changed.